app/registries/foreman/access_control.rb

```diff
@@ -22,7 +22,14 @@ def map
         mapper = Mapper.new
         yield mapper
         @permissions ||= []
-        @permissions += mapper.mapped_permissions
+        mapper.mapped_permissions.each do |permission|
+          if (perm = permission(permission.name))
+            Rails.logger.warn "You are trying to replace #{perm.name} from #{permission.engine}. Adding actions from plugin permissions to a core one."
```

Comment:
logger should be enough I think
```suggestion
            logger.warn "You are trying to replace #{perm.name} from #{permission.engine}. Adding allowed actions from plugin permissions to the existing one."
```